numba/targets/arraymath.py

```diff
@@ -3097,6 +3097,20 @@ def diff_impl(a, n=1):
     return diff_impl
 
 
+@overload(np.array_equal)
+def np_array_equal(a, b):
+
+    if not (isinstance(a, types.Array) and isinstance(b, types.Array)):
```

Comment:
Despite having "array" in the function name, this function seems to accept the usual selection of array-like things.
```pycon
In [1]: import numpy as np                                                                                                      

In [2]: np.array_equal((1,2,3),(1,2,3))                                                                                         
Out[2]: True

In [3]: np.array_equal(1,2)                                                                                                     
Out[3]: False

In [4]: np.array_equal(1,1)                                                                                                     
Out[4]: True
```